packages/server/src/orchestrator/atom.ts

```diff
@@ -36,7 +38,16 @@ export const createOrchestratorAtom = (project: OrchestratorAtomOptions) => {
       options: project.app,
     },
     proxyService: {
-      proxyStatus: 'unstarted'
+      status: {type: 'unstarted'},
+      options: {
+        port: project.proxy.port,
+        prefix: agentServerConfig.options.prefix,
+        appDir: agentServerConfig.appDir(),
+        harnessUrl: agentServerConfig.harnessUrl(),
+        // TODO: this is duplication because currently we can only pass 1 slice into a service
+        // Is this problematic?
+        appOptions: project.app
```

Comment:
I'm not super concerned about finding away around this duplication for now as I think there are a number of strategies we can follow and I'm not sure which is the best one.

However, on the set side, maybe we can hedge by instead of having the test helper set both locations, but instead just have an operation in the orchestrator that syncs one to the other. Roughly:

```ts
atom.slice('appOptions').forEach(state => {
  atom.slice('proxyService','options','appOptions').set(state);
});
```

That way, if we ever do get to a point where we're setting the app options from a running orchestrator it will work in both the deployed version and the test case.

Reply:
Maybe we could have one unified options property in the state, which would store all of the options in the same format that we have in the `json` file?

Reply:
(not saying we should do this in this PR, but as a future direction to head in)

Reply:
@cowboyd @jnicklas I was heading in the direction of keeping all configuration out of orchestrator.  By the time it reaches there, it should be just a matter of starting the services in the correct sequence.  

I quite liked the idea of each service having its own config and a status.

A service or really anything to do with the state should only care about the information it needs in order for it to do what it has to do.

The `proxyService` blew that out of the water, unfortunately.

> However, on the set side, maybe we can hedge by instead of having the test helper set both locations

Is keeping it in sync only important for the tests?  If so then I would be hesitant to do anything in the production code, just to make the tests pass.

Reply:
fwiw, I think the proxyService and `appService` will end up being merged since they're really part of the same mechanics of testing webapps.

Reply:
Which is to say I expect they'll get "de-duped" later on down the road...

Reply:
@cowboyd it looks like the `url` is the only property it needs from `appOptions`.

Am I right in saying that the point of 

```ts
return restartable(appOptions, startProxyServer(serviceSlice));` 
```

is to restart if any of the props of the slice change?

If so then maybe we could put something in the orchestrator to sync this like you mentioned earlier.

Reply:
or maybe as @jnicklas we could have all the config in one slice and restart when anything changes.

Reply:
are we happy just to live with the duplication for now?